pkg/storage/batch.go

```diff
@@ -723,6 +735,17 @@ func fetchLazyChunks(ctx context.Context, s chunk.SchemaConfig, chunks []*LazyCh
 				return
 
 			}
+			if postFetcherChunkFilterer != nil {
+				for i, chk := range chks {
+					chunks[i].Chunk = chk
+				}
```

Comment:
I think we might not get the chunks back in the same order due to two reasons:
* `postFetchFilter` might drop some of the chunks due to none of its line matching the expression.
* As mentioned below, we don't get chunks back in the same order from `FetchChunks`.

Reply:
ok,I will try fix this.

Reply:
```go
fix. 
for i, chk := range chks {
    key := chunkKeys[i]
    index[key].Chunk = chk
}
```